concordium-node/src/consensus_ffi/ffi.rs

```diff
@@ -2195,6 +2215,30 @@ impl ConsensusContainer {
         Ok(out_data)
     }
 
+    pub fn get_consensus_detailed_status_v2(
+        &self,
+        genesis_index: Option<u32>,
+    ) -> Result<Vec<u8>, tonic::Status> {
+        let consensus = self.consensus.load(Ordering::SeqCst);
+        let mut out_data: Vec<u8> = Vec::new();
+        let (use_genesis_index, genesis_index) = match genesis_index {
+            Some(genesis_index) => (1, genesis_index),
+            None => (0, 0),
+        };
+        let response: ConsensusQueryResponse = unsafe {
```

Comment:
what's the reason for the `unsafe`?

Reply:
FFI calls are unsafe in Rust (i.e. they can break type safety). `unsafe` tells the compiler to trust that we know this particular code won't break safety.